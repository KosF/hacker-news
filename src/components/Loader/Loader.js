import React from "react";

import LoaderIcon from "./Loader.styles";

const Loader = () => <LoaderIcon />;

export default Loader;
