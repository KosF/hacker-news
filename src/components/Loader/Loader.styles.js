import styled, { keyframes } from "styled-components";

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const LoaderIcon = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -40px;
  border: 8px solid #f3f3f3;
  border-top: 8px solid #f60;
  border-radius: 50%;
  width: 80px;
  height: 80px;
  animation: ${rotate} 2s linear infinite;
`;

export default LoaderIcon;
