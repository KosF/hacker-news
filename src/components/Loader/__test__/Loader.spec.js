import React from "react";

import Loader from "../Loader";

describe("<Loader />", () => {
  const wrapper = global.mount(<Loader />);

  it("should contains loader icon", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
