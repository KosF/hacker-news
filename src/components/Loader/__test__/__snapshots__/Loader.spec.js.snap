// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should contains loader icon 1`] = `
<Loader>
  <styled.span>
    <span
      className="sc-AxjAm cqrqRC"
    />
  </styled.span>
</Loader>
`;
